### ISSUE TITLE
<Link>s has to been pressed twice on an iOS-Device with a ScrollUp-Button

### ISSUE THREAD

**Issue by t-kietzmann** [gatsbyjs/gatsby#36952]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

Hey all, after i spend tons of hours to find the reason why i have to click on some <Link>s twice on an iOS-Device, i found this out:

When using a ScrollToTop Component with an useEffect and a dynamic "Up-Button" you have to press a <Link> twice on iOS to navigate. 

This happens only on iOS-Devices (you can test it also at the Simulator).
I'm trying to find out more about what's causing it.

### Reproduction Link

https://github.com/t-kietzmann/gatsby-link-bug

### Steps to Reproduce

1. Check https://github.com/t-kietzmann/gatsby-link-bug
2. Click the first button, it will work with one click on iOS
3. Scroll down (the UP must be visible in the right bottom corner, maybe you have to shrink your window)
4. Click the button, nothing will happen on the first click.

### Expected Result

The <Link> should work after one click

### Actual Result

The <Link> works only after two clicks

### Environment

```shell
Gatsby 4.24.6
Every iPhone
```

**Comment by LekoArts** [comment]
While I do see the behavior you're describing in your reproduction, I'm not convinced that this is a bug with Gatsby. In your reproduction I also have to double click the "Go to second with visible UP button"-button. And if that were the case for every Gatsby website we'd already have heard about this. For example if I go to my website (https://www.lekoarts.de) and scroll down a bit, every link just works on the first click.

The clicks work on the first try if the browser section of iOS (the bottom bar with the navigation, export, etc. is visible. If you also Google for it you'll find:

- https://css-tricks.com/annoying-mobile-double-tap-link-issue/
- https://stackoverflow.com/questions/66906513/have-to-tap-twice-on-bottom-part-in-ios-safari-browser
